examples/server_client.rs

```diff
@@ -0,0 +1,100 @@
+//! Note that the terms "client" and "server" here are purely what we logically associate with them.
+//! Technically, they both work the same.
+
+extern crate laminar;
+
+use std::{
+    io::stdin,
+    net::{SocketAddr, ToSocketAddrs},
+};
+
+use laminar::{
+    error::Result,
+    Packet,
+    UdpSocket,
+};
+
+const SERVER: &str = "localhost:12351";
+
+fn addr(s: &str) -> Result<SocketAddr> {
+    s.to_socket_addrs().map_err(Into::into).map(|mut x| x.next().unwrap())
+}
+
+fn server() -> Result<()> {
+    let mut socket = UdpSocket::bind(SERVER)?;
+
+    loop {
+        match socket.recv()? {
+            Some(packet) => {
+                let msg = packet.payload();
+
+                if msg == b"Bye!" {
+                    break;
+                }
+
+                let msg = String::from_utf8_lossy(msg);
+                let ip = packet.addr().ip();
+
+                println!("Received {:?} from {:?}", msg, ip);
+
+                socket.send(Packet::new(packet.addr(), "Copy that!".as_bytes().to_vec()))??;
+            }
+            None => {}
+        }
+    }
+
+    Ok(())
+}
+
+fn client() -> Result<()> {
+    let mut socket = UdpSocket::bind("localhost:12352")?;
+
+    let server = addr(SERVER)?;
+
+    println!("Type a message and press Enter to send. Send `Bye!` to quit.");
+
+    let stdin = stdin();
+    let mut s = String::new();
+
+    loop {
+        s.clear();
+        stdin.read_line(&mut s)?;
+        let line = s.replace(|x| x == '\n' || x == '\r', "");
+
+        socket.send(Packet::new(server, line.clone().into_bytes()))??;
```

Comment:
`??`?

Reply:
Yes, those are result wrapped results, most likely for non-blocking stuff.